File: VB/DateAndTimeColumns_Unbound/MainViewModel.vb
Imports DevExpress.Mvvm
Imports System
Imports System.Collections.ObjectModel
Imports System.Linq

Namespace DateAndTimeColumns_Unbound

    Friend Class MainViewModel
        Inherits BindableBase

        Public ReadOnly Property Source As ObservableCollection(Of Item)

        Public Sub New()
            Source = New ObservableCollection(Of Item)(Enumerable.Range(0, 1000).[Select](Function(i) New Item With {.DateTime = New DateTime(1500 + i, 1 + i Mod 12, 1 + i Mod 28, i Mod 24, i Mod 60, i Mod 60)}))
        End Sub
    End Class
End Namespace

File: CS/DateAndTimeColumns_Properties/MainViewModel.cs
using DevExpress.Mvvm;
using System;
using System.Collections.ObjectModel;
using System.Linq;

namespace DateAndTimeColumns_Properties
{
    internal class MainViewModel : BindableBase
    {
        public ObservableCollection<Item> Source { get; }

        public MainViewModel() {
            Source = new ObservableCollection<Item>(Enumerable.Range(0, 1000).Select(i => new Item {
                DateTime = new DateTime(1500 + i, 1 + i % 12, 1 + i % 28, i % 24, i % 60, i % 60)
            }));
        }
    }
}
